hikari/impl/rest.py

```diff
@@ -551,74 +553,74 @@ async def _request(
 
         headers.put(_X_AUDIT_LOG_REASON_HEADER, reason)
 
+        url = compiled_route.create_url(self._rest_url)
         while True:
-            try:
-                url = compiled_route.create_url(self._rest_url)
-
-                # Wait for any rate-limits to finish.
-                await asyncio.gather(self.buckets.acquire(compiled_route), self.global_rate_limit.acquire())
-
-                uuid = time.uuid()
-
-                if _LOGGER.getEffectiveLevel() <= ux.TRACE:
-                    _LOGGER.log(
-                        ux.TRACE,
-                        "%s %s %s\n%s",
-                        uuid,
+            async with self._lock:
```

Comment:
could we be more specific about when we aquire this lock?

Reply:
I decided to implement a per bucket lock to stop 2 requests going towards the same bucket at the same time. Doing that rn, will update pr when im done :+1: